### PR TITLE
Add enforce release label workflow

### PR DESCRIPTION
**Pull request by joshkmartinez**
## Description
Adds a Github workflow to enforce release labels on PRs into the main branch.

## Demos
Works on a PR into a test branch in https://github.com/chanzuckerberg/napari-hub/pull/1143#issuecomment-1634975429

### CODE REVIEW

File: .github/workflows/enforce-labels.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+      - name: Install dependencies
+        run: npm install @octokit/rest @actions/github
+
+      - name: Check for labels
+        run: |
+          node .github/check-labels.js
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (joshkmartinez) [reply .github/workflows/enforce-labels.yml]: Was able to shorten it. Here's it working on another test PR: https://github.com/chanzuckerberg/napari-hub/pull/1145#issuecomment-1635044816

**Comment by joshkmartinez** [comment]
Adresses #1146

File: .github/check-labels.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    repo: context.repo.repo,
+    issue_number: pr.number,
+  });
+
+  const labelNames = labels.data.map((label) => label.name);
+  const configLabels = config.categories.flatMap((category) => category.labels);
+
+  const hasRequiredLabel = labelNames.some((name) => configLabels.includes(name));
+
+  if (!hasRequiredLabel) {
```
Comment (manasaV3) [inline .github/check-labels.js:22]: We should add an else case, where we check if the comment below exists, and if it does either update the comment or delete it.

Comment (joshkmartinez) [reply .github/check-labels.js:22]: Nice idea! Just opened up #1150 for this

**Pull request merged**